lianad/src/database/mod.rs

```diff
@@ -484,6 +493,26 @@ impl DatabaseConnection for SqliteConn {
             .collect()
     }
 
+    fn insert_input_seen_before(&mut self, outpoints: &[bitcoin::OutPoint]) -> bool {
+        let pj_outpoints = self.outpoints_from_payjoin(outpoints);
+        let was_seen_before = outpoints
+            .iter()
+            .any(|outpoint| pj_outpoints.contains_key(outpoint));
+        self.insert_outpoint_seen_before(pj_outpoints.keys());
+
+        was_seen_before
+    }
+
+    fn outpoints_from_payjoin(
+        &mut self,
+        outpoints: &[bitcoin::OutPoint],
+    ) -> HashMap<bitcoin::OutPoint, Option<u32>> {
+        self.get_seen_payjoin_outpoints(outpoints)
+            .into_iter()
+            .map(|db_pjoutpoint| (db_pjoutpoint.outpoint, db_pjoutpoint.added_at))
```

Comment:
Nit. This var is a bit hard to read. I would just go for ot.
```suggestion
            .map(|ot| (ot.outpoint, ot.added_at))
```